### PR TITLE
refactor: remove scaling in apportionment, simplify algorithm

### PR DESCRIPTION
**Pull request by karlwuest**
## Description

Removes scaling of the stake for the apportionment algorithm and simplifies the algorithm itself by using a custom max heap instead of the priority queue from the sui-framework. Also reduces the gas cost for the apportionment a bit as a side effect. 

Closes WAL-443
Closes WAL-211

## Test plan

Added tests for new queue, existing tests still test the high level behaviour of the apportionment.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.
For each box you select, include information after the relevant heading that describes the impact of your changes that
a user might notice and any actions they must take to implement updates. (Add release notes after the colon for each item)

- [ ] Storage node:
- [ ] Aggregator:
- [ ] Publisher:
- [ ] CLI:

### CODE REVIEW

File: contracts/walrus/sources/staking/apportionment_queue.move [hunk 1]
```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// A custom priority queue implementation for use in the apportionment algorithm.
+/// This implementation uses a quotient-based priority with a tie-breaker to break ties when
+/// priorities are equal.
+module walrus::apportionment_queue;
```
Comment (tnowacki) [inline contracts/walrus/sources/staking/apportionment_queue.move:7]: @cgswords @tzakian @dariorussi @awelc.
Food for thought here when thinking about signatures next year. It's a good one.

File: contracts/walrus/sources/staking/uq64_64.move
Comment (karlwuest) [inline contracts/walrus/sources/staking/uq64_64.move:1]: TODO: remove and replace all occurences with `std::uq64_64` once Sui `testnet-v1.39.0` is released.

Comment (mlegner) [reply contracts/walrus/sources/staking/uq64_64.move:1]: I guess this means that this whole file will go away, correct?

Comment (karlwuest) [reply contracts/walrus/sources/staking/uq64_64.move:1]: Yes, correct.

Comment (karlwuest) [reply contracts/walrus/sources/staking/uq64_64.move:1]: Done

**Pull request merged**